src/test/java/com/commercetools/sync/commons/MockUtils.java

```diff
@@ -44,6 +44,33 @@ public static CustomFieldsDraft getMockCustomFieldsDraft() {
         return CustomFieldsDraft.ofTypeIdAndJson("StepCategoryTypeId", customFieldsJsons);
     }
 
+    /**
+     * Returns mock {@link CustomFields} instance. Executing {@link CustomFields#getType()} on returned instance will
+     * return {@link Reference} of given {@code typeId} with mock {@link Type} instance of {@code typeId} and {@code
+     * typeKey} (getters of key and id would return given values). Executing {@link CustomFields#getFieldsJsonMap()} on
+     * returned instance will return {@link Map} populated with given {@code fieldName} and {@code fieldValue}
+     *
+     * @param typeId custom type id
+     * @param fieldName custom field name
+     * @param fieldValue custom field value
+     * @return mock instance of {@link CustomFields}
+     */
+    public static CustomFields getMockCustomFields(final String typeId, final String fieldName,
+                                                   final JsonNode fieldValue) {
+        final CustomFields customFields = mock(CustomFields.class);
```

Comment:
Why mocking rather than returning the actual `CustomFields` instance? are you going to test the behavior of the mock? 

I only see asserts for facts but not for behaviors in your tests.

Reply:
Because the `CustomFields` cannot be instantiated through our JVM SDK, as it's treated as a read-only model, unlike `CustomFieldsDraft` which is the read-write model that can actually be instantiated.